src/curl_async.c

```diff
@@ -453,7 +453,7 @@ static int process_download(struct swupd_curl_parallel_handle *h, struct multi_c
 {
 	CURL *curl = NULL;
 	CURLMcode curlm_ret = CURLM_OK;
-	CURLcode curl_ret = CURLE_OK;
+	CURLcode curl_ret = -1;
```

Comment:
Try not to mix different types. Use only Curl code for Curl errors.
The ideal too would be to return `CURLcode` in this function too

Reply:
I used -1 because the function already returned -1 on any error before, but I agree it is better not to mix types

Reply:
I'll change that